modules/django-openai/django_openai/openai/viewsets.py

```diff
@@ -0,0 +1,138 @@
+        """
+        To get the models details
+        :path_params: model_id
+        :return: Returns models details.
+        """
+        response = self.openai_service.retrieve_model(model_id)
+        return Response(data=response.get("data"), status=response.get("status_code"))
+
+    @action(detail=False, methods=['get'], url_path='get_engine_list')
+    def get_engine_list(self, request):
+        """
+        To get the engine list
+        :return: Returns list of engine.
+        """
+        response = self.openai_service.list_engine()
+        return Response(data=response.get("data"), status=response.get("status_code"))
+
+    @action(detail=False, methods=['get'], url_path='get_engine_detail/(?P<engine_id>[^/.]+)')
+    def get_engine_detail(self, request, engine_id):
+        """
+        To get the engine details'
+        :path_params: engine_id
+        :return: Returns models details.
+        """
+        response = self.openai_service.retrieve_engine(engine_id)
+        return Response(data=response.get("data"), status=response.get("status_code"))
+
+    @action(detail=False, methods=['post'], url_path='create_chat_completion')
+    def create_a_chat_completion(self, request):
+        """
+        To create a chat completion
+        :payload: {"model": "", "messages": ["role": "", "content": ""]}
+        :return: Returns a newly created chat completion detail.
+        """
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        response = self.openai_service.create_chat_completion(payload=serializer.data)
+        return Response(data=response.get("data"), status=response.get("status_code"))
+
+    @action(detail=False, methods=['post'], url_path='create_a_completion')
+    def create_a_completion(self, request):
+        """
+        To create a completion
+        :payload: {"model": "", "prompt": "", "max_tokens": "", "temperature": ""}
+        :return: Returns a newly created completion.
+        """
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        response = self.openai_service.create_completion(payload=serializer.data)
+        return Response(data=response.get("data"), status=response.get("status_code"))
+
+    @action(detail=False, methods=['post'], url_path='create_an_edit')
+    def create_an_edit(self, request):
+        """
+        To create an edit
+        :payload: {"model": "", "input": "", "instruction": ""}
+        :return: Returns a newly created edit with detail.
+        """
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        response = self.openai_service.create_edit(payload=serializer.data)
+        return Response(data=response.get("data"), status=response.get("status_code"))
+
+    @action(detail=False, methods=['post'], url_path='create_an_image')
```

Comment:
lets keep the name of urls same a openAI urls for all of the ulrs and keep REST standards in mind not user readability 
keep these closer to these ```f'{self.OPENAI_BASE_URL}/v1/images/generations'```